### DISCUSSION TITLE
Implement Next-Auth to consume protected routes of custom Laravel Backend

### DISCUSSION THREAD

**Discussion by Aniket-IN** [nextauthjs/next-auth#6338] [Help]
Hi, 
I'm very new to Next-Auth.

### Here's my tech stack: 
- [Laravel](https://laravel.com/) as the Backend
- [Laravel Passport](https://laravel.com/docs/9.x/passport) for OAuth 
- Next.js 13 (`App Directory`)
- Next-Auth `4.18.8`

### What I'm trying to achieve?
I already have the backend developed in `Laravel`, I wan to use `Next.js` just for serving to the end user.
I've configured custom OAuth provider in `pages/api/auth/[...nextauth].ts` like this:
```js
import NextAuth, { NextAuthOptions } from "next-auth"
export const authOptions: NextAuthOptions = {
  providers: [
    {
      id: "I4T",
      name: "I4T",
      type: "oauth",
      authorization: {
        url: `${process.env.I4T_BACKEND_URL}/oauth/authorize`,
        params: { scope: "" }
      },
      token: `${process.env.I4T_BACKEND_URL}/oauth/token`,
      userinfo: `${process.env.I4T_BACKEND_URL}/api/user`,
      clientId: process.env.I4T_CLIENT_ID,
      clientSecret: process.env.I4T_CLIENT_SECRET,
      profile: (profile) => {
        return profile
      },
      
    }
  ],
}
export default NextAuth(authOptions)
```
It works as expected. 

But I also want to call any subsequent protected API routes (_e.g: perform some CRUD operations etc.._) of the Laravel backend from the Next.js application (both `Client components` and `Server components`) and for example if any of the call returns `Unauthorized` I want the Next.js application to be logged out.

I'm not sure how to achieve the same in most secure way, read many articles/tutorials but all of them ends at just setting up custom OAuth provider. :(

Any help is much appricated :)

**Comment by nobilelucifero** [comment]
@Aniket-IN any luck with it?

**Comment by clicktodev** [comment]
This has a possible solution.
https://github.com/nextauthjs/next-auth/discussions/4900#discussioncomment-3122781

**Comment by alea12** [comment]
@Aniket-IN sorry to be off-topic, but I'm now looking for a way to implement `userinfo` endpoint in the Laravel Passport side. Could you guide me to some references if possible?
